src/lib/util/Timestamp.js

```diff
@@ -1,5 +1,79 @@
+	['hh', time => String(time.getHours() % 12 || 12).padStart(2, '0')],
+	['a', time => time.getHours() < 12 ? 'am' : 'pm'],
+	['A', time => time.getHours() < 12 ? 'AM' : 'PM'],
+	['m', time => String(time.getMinutes())],
+	['mm', time => String(time.getMinutes()).padStart(2, '0')],
+	['s', time => String(time.getSeconds())],
+	['ss', time => String(time.getSeconds()).padStart(2, '0')],
+	['S', time => String(time.getMilliseconds())],
+	['SS', time => String(time.getMilliseconds()).padStart(2, '0')],
+	['SSS', time => String(time.getMilliseconds()).padStart(3, '0')],
+	['T', time => `${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')} ${time.getHours() < 12 ? 'AM' : 'PM'}`],
+	['t', time => `${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')}:${String(time.getSeconds()).padStart(2, '0')} ${time.getHours() < 12 ? 'am' : 'pm'}`],
+	['L', time => `${String(time.getMonth() + 1).padStart(2, '0')}/${String(time.getDate()).padStart(2, '0')}/${String(time.getFullYear())}`],
+	['l', time => `${String(time.getMonth() + 1)}/${String(time.getDate()).padStart(2, '0')}/${String(time.getFullYear())}`],
+	['LL', time => `${MONTHS[time.getMonth()]} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())}`],
+	['ll', time => `${MONTHS[time.getMonth()].slice(0, 3)} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())}`],
+	['LLL', time => `${MONTHS[time.getMonth()]} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())} ${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')} ${time.getHours() < 12 ? 'AM' : 'PM'}`],
+	['lll', time => `${MONTHS[time.getMonth()].slice(0, 3)} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())} ${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')} ${time.getHours() < 12 ? 'AM' : 'PM'}`],
+	['LLLL', time => `${DAYS[time.getDay()]}, ${MONTHS[time.getMonth()]} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())} ${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')} ${time.getHours() < 12 ? 'AM' : 'PM'}`],
+	['llll', time => `${DAYS[time.getDay()].slice(0, 3)} ${MONTHS[time.getMonth()].slice(0, 3)} ${String(time.getDate()).padStart(2, '0')}, ${String(time.getFullYear())} ${String(time.getHours() % 12 || 12)}:${String(time.getMinutes()).padStart(2, '0')} ${time.getHours() < 12 ? 'AM' : 'PM'}`],
+	['Z', time => {
+		const offset = time.getTimezoneOffset();
+		return `${offset >= 0 ? '+' : '-'}${String(offset / -60).padStart(2, '0')}:${String(offset % 60).padStart(2, '0')}`;
+	}],
+	['ZZ', time => {
+		const offset = time.getTimezoneOffset();
+		return `${offset >= 0 ? '+' : '-'}${String(offset / -60).padStart(2, '0')}:${String(offset % 60).padStart(2, '0')}`;
+	}]
+]);
+/* eslint-enable max-len */
+
+const tokenRepeatingCounts = new Map(Object.entries(TOKENS));
```

Comment:
We only use TOKENS for this, if you want it to be a map, make it a map to start with instead of doubling up the memory footprint.

Reply:
Addressed in 04ab830